text/1071-cell.md

```diff
@@ -0,0 +1,296 @@
+project-link: Leave as is
+suite: Leave as is
+-->
+
+<-- Replace "RFC title" with the title of your RFC -->
+# Introduce `Cell` 
+
+## Summary
+
+This RFC introduces a new tracking primitive, which represents a single value, the `Cell`.
+
+## Motivation
+
+The `Cell` is part of the "spreadsheet analogy" when talking about reactivity -- it represents a single tracked value, and can be created without the use of a class, making it a primate candidate for demos[^demos] and for creating reactive values in function-based APIs, such as _helpers_, _modifiers_, or _resources_. They also provide a benefit in testing as well, since tests tend to want to work with some state, the `Cell` is wholly encapsulated, and can be quickly created with 0 ceremony. 
+
+This is not too dissimilar to the [Tracked Storage Primitive in RFC#669](https://github.com/emberjs/rfcs/blob/master/text/0669-tracked-storage-primitive.md). The `Cell` provides more ergonomic benefits as it doesn't require 3 imports to use. 
+
+The `Cell` was prototyped in [Starbeam](https://starbeamjs.com/guides/fundamentals/cells.html) and has been available for folks to try out in ember via [ember-resources](https://github.com/NullVoxPopuli/ember-resources/tree/main/docs/docs). 
+
+[^demos]: demos _must_ over simplify to bring attention to a specific concept. Too much syntax getting in the way easily distracts from what is trying to be demoed. This has benefits for actual app development as well though, as we're, by focusing on concise demo-ability, gradually removing the amount of typing needed to create features. 
+
+## Detailed design
+
+
+### Types 
+
+Some interfaces to share with future low-level reactive primitives:
+
+```ts
+
+interface Reactive<Value> {
+    /**
+    * The underlying value
+    */
+    current: Value;
+    /**
+    * Returns the underlying value
+    */
+    read(): Value;
+}
+
+interface ReadOnlyReactive<Value> extends Reactive<Value> {
+    /**
+    * The underlying value.
+    * Cannot be set.
+    */
+    readonly current: Value;
+
+    /**
+    * Returns the underlying value
+    */
+    read(): Value;
+}
+
+interface Cell<Value> extends Reactive<Value> {
+    /**
+    * Utility to create a Cell without the caller using the `new` keyword.
+    */
+    static create<T>(initialValue: T, equals?: (a: T, b: T) => boolean): Cell<T>;
+
+    /**
+    * The constructor takes an optional initial value and optional equals function.
+    */
+    constructor(initialValue?: Value, equals?: (a: Value, b: Value) => boolean): {}
+
+    /**
+    * Function short-hand of updating the current value
+    * of the Cell
+    */
+    set: (value: Value) => boolean;
+    /**
+    * Function short-hand for using the current value to 
+    * update the state of the Cell
+    */
+    update: (fn: (value: Value) => Value) => void;
+
+    /**
+     * Prevents further updates, making the Cell
+     * behave as a ReadOnlyReactive
+     */
+    freeze: () => void;
```

Comment:
For "feature parity" with `Object.freeze`, would it make sense to also have `isFrozen()`, as in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/isFrozen?

Reply:
I don't think we need to worry about checking this -- it's an optimization for caches'n' such

Reply:
Tho, if you or anyone can think of a good use case for isFrozen, maybe we add it?